contracts/0.8.9/CommitteeQuorum.sol

```diff
@@ -80,66 +86,87 @@ contract CommitteeQuorum {
         if (bitMask & mask != 0) { revert MemberAlreadyReported(); }
         REPORTS_BITMASK_POSITION.setStorageUint256(bitMask | mask);
 
-        bytes32 reportHash = keccak256(_report);
-        isQuorumReached = false;
+        uint256 numDistinctReports = distinctReportHashes.length;
 
         uint256 i = 0;
         bool isFound = false;
-        while (i < distinctReports.length && distinctReportHashes[i] != reportHash) {
+        while (i < numDistinctReports && distinctReportHashes[i] != _reportHash) {
             ++i;
         }
-        while (i < distinctReports.length) {
-            if (distinctReportHashes[i] == reportHash) {
+        while (i < numDistinctReports) {
+            if (distinctReportHashes[i] == _reportHash) {
```

Comment:
IDK, check, please, but this `if` looks like always true for me. So while will run exactly once, if run at all.

Reply:
refactored, thanks